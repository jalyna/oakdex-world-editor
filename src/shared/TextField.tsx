import * as React from 'react'
import styled from 'styled-components'

import { DEFAULT_FONT, GREY_30, GREY_70 } from 'shared/theme'

const StyledTextField = styled.input`
  font-family: ${DEFAULT_FONT};
  padding: 5px 10px;
  color: ${GREY_30};
  border: 1px solid ${GREY_70};
  font-size: 16px;
  outline: none;
  margin: 0;
  background: white;
  box-sizing: border-box;
  width: 100%;
`

export default function TextField (props: any) {
  return <StyledTextField {...props} />
}
