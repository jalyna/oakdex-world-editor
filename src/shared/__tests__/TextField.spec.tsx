import * as React from 'react'
import { mount } from 'enzyme'
import TextField from '../TextField'

describe('<TextField>', () => {
  it('renders correctly', () => {
    const wrapper = mount(<TextField value='test' onChange={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
})
