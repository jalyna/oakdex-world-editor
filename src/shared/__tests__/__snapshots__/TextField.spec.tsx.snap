// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField> renders correctly 1`] = `
<TextField
  onChange={[MockFunction]}
  value="test"
>
  <styled.input
    onChange={[MockFunction]}
    value="test"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "dfAss",
            "rules": Array [
              "
  font-family: ",
              "'Roboto', sans-serif",
              ";
  padding: 5px 10px;
  color: ",
              "#4d4d4d",
              ";
  border: 1px solid ",
              "#b3b3b3",
              ";
  font-size: 16px;
  outline: none;
  margin: 0;
  background: white;
  box-sizing: border-box;
  width: 100%;
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[MockFunction]}
      value="test"
    >
      <input
        className="sc-bdVaJa dfAss"
        onChange={[MockFunction]}
        value="test"
      />
    </StyledComponent>
  </styled.input>
</TextField>
`;
